src/finiteVolume/fluxCalculator/riemannCommon.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef ABLATELIBRARY_RIEMANNCOMMON_HPP
+#define ABLATELIBRARY_RIEMANNCOMMON_HPP
+
+#include "fluxCalculator.hpp"
+/**
+ * Solve the Riemann problem.
+ * @param uL: velocity on the left cell center
+ * @param aL: speed of sound on the left center cell
+ * @param rhoL: density on the left cell center
+ * @param p0L: reference pressure for stiffened gas on left (pass in from EOS). Will be zero if perfect gas or single gas
+ * @param pL: pressure on the left cell center
+ * @param gammaL: specific heat ratio for gas on left (pass in from EOS)
+ * @param uR: velocity on the right cell center
+ * @param aR: speed of sound on the right center cell
+ * @param rhoR: density on the right cell center
+ * @param p0R: reference pressure for stiffened gas on right (pass in from EOS). Will be zero if perfect gas or single gas
+ * @param pR: pressure on the right cell center
+ * @param gammaR: specific heat ratio for gas on right (pass in from EOS)
+ * @param pstar0: initial guess at the pressure across contact surface
+ * @param massFlux: mass flux across the cell face(?)
+ * @param p12: interface pressure(?)
+ */
+ablate::finiteVolume::fluxCalculator::Direction riemannSolver(const PetscReal uL, const PetscReal aL, const PetscReal rhoL, const PetscReal p0L, const PetscReal pL, const PetscReal gammaL,
```

Comment:
For something like we I think we might want to keep riemannSolver function out of the global name space.  Would you mind if we moved it to a static function inside of Riemann?

Reply:
This is just to get access to the direction enum. The file riemann.cpp contains the single ideal gas. Is this where you were thinking of moving it to?

Reply:
Sure. I can make the Riemann solver a sub-class of fluxCalculator and the three cases sub-classes of the Riemann solver.